osf/utils/workflows.py

```diff
@@ -355,6 +378,81 @@ def db_name(self):
     },
 ]
 
+
+COLLECTION_SUBMISSION_TRANSITIONS = [
+    {
+        'trigger': 'submit',
+        'source': [CollectionSubmissionStates.IN_PROGRESS],
+        'dest': CollectionSubmissionStates.ACCEPTED,
+        'before': [],
+        'after': [],
+        'unless': ['is_moderated'],
+    },
+    {
+        'trigger': 'submit',
+        'source': [CollectionSubmissionStates.IN_PROGRESS],
+        'dest': CollectionSubmissionStates.PENDING,
+        'before': [],
+        'after': ['_notify_moderated_pending'],
+        'conditions': ['is_moderated'],
+    },
+    {
+        'trigger': 'accept',
+        'source': [],
+        'dest': None,
+        'before': [],
+        'after': [],
+        'unless': ['is_moderated'],
+    },
+    {
+        'trigger': 'accept',
+        'source': [CollectionSubmissionStates.PENDING],
+        'dest': CollectionSubmissionStates.ACCEPTED,
+        'before': ['_validate_accept'],
+        'after': ['_notify_moderated_accepted'],
+        'conditions': ['is_moderated'],
+    },
+    {
+        'trigger': 'reject',
+        'source': [],
+        'dest': None,
+        'before': [],
+        'after': [],
+        'unless': ['is_moderated'],
+    },
```

Comment:
Would consider omitting these so we just get a MachineError on an `accept`/`reject` on an unmoderated submission?